mkdocs/serve.py

```diff
@@ -71,7 +71,7 @@ def builder():
             theme=theme,
         )
         config['site_dir'] = tempdir
-        build(config, live_server=True)
+        build(config, live_server=True, clean_site_dir=True)
```

Comment:
We remove the tempdir when the server shutdown already. https://github.com/mkdocs/mkdocs/blob/master/mkdocs/serve.py#L88

This is to do with multiple builds happening within one "session" of running `mkdocs serve`. We could create a new tempdir on each build and change where we are serving from... but that sounds like hard work.